src/scan_parallel_inclusive/scan.cpp

```diff
@@ -70,14 +70,14 @@ void par_scan(sycl::buffer<T, 1>& in, sycl::queue& q) {
 
   // Check if there is enough global memory.
   size_t global_mem_size = dev.get_info<sycl::info::device::global_mem_size>();
-  if (!dev.is_host() && in.get_count() > (global_mem_size / 2)) {
+  if (in.size() > (global_mem_size / 2)) {
     throw std::runtime_error("Input size exceeds device global memory size.");
   }
 
   /* Check if local memory is available. On host no local memory is fine, since
```

Comment:
Can you edit this comment as well, to remove mention of the host device?

Reply:
Good catch, updated.